app/controllers/flood_risk_engine/enrollments/addresses_controller.rb

```diff
@@ -48,7 +48,7 @@ def failure_url
             enrollment,
             address_initial_attributes.merge(
               check_for_error: true,
-              step_back_to_postcode: params[:step_back_to_postcode]
+              step_back_to_postcode: params[:step_back_to_postcode], target_fwd_state: params[:target_fwd_state]
```

Comment:
I know this is picky, but could the arrangement of attributes please be consistent. So
```
address_initial_attributes.merge(
  check_for_error: true,
  step_back_to_postcode: params[:step_back_to_postcode], 
  target_fwd_state: params[:target_fwd_state]
)
```